tools/monitoring/cloudmail/mail.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudmail
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	mail "cloud.google.com/go/mail/apiv1alpha3"
+	mailpb "google.golang.org/genproto/googleapis/cloud/mail/v1alpha3"
+	"google.golang.org/genproto/protobuf/field_mask"
+)
+
+const (
+	projectID = "knative-tests"
+	region    = "us-central1"
+
+	addressSetID   = "monitoring-address-set"
+	addressPattern = "monitoring-alert"
+	senderID       = "monitoring-alert-sender"
+	receiptRuleID  = "monitoring-receipt-drop-rule"
+)
+
+// MailClient holds instance of CloudMailClient for making mail-related requests
+type MailClient struct {
+	CloudMailClient *mail.CloudMailClient
```

Comment:
can we create an embedded type instead of a new field?

```
type MailClient struct {
	*mail.CloudMailClient
	DomainName      string
	DomainID        string
}
```

you can set as per https://stackoverflow.com/questions/27626765/mixture-of-fieldvalue-and-value-initializers

Reply:
Also, do we need to export domainname and domainid or is it only for this wrapper?

Reply:
It is only for this wrapper. I thought of using constants and set these fields at the top, but that would prevent us from having the one step setup that creates both the project domain as well as the sender. 

However, I'm not sure if the one step setup is valuable enough to add those fields in the struct though. The reason is that we can only set up one project domain per gcp project.So that function will only ever be run once. To test that flow works, I actually created a new GCP project to test it. Furthermore, I don't think we will use a custom domain with the current mail client in the future. Since if we have a custom email domain and can already authenticate to that, we can use [golang smtp](https://golang.org/pkg/net/smtp/) directly instead of going through cloud mail.